src/visitor.rs

```diff
@@ -592,16 +592,24 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
         self.push_rewrite_inner(span, rewrite);
     }
 
-    pub fn push_skipped_with_span(&mut self, attrs: &[ast::Attribute], item_span: Span) {
+    pub fn push_skipped_with_span(
+        &mut self,
+        attrs: &[ast::Attribute],
+        item_span: Span,
+        main_span: Span,
+    ) {
         self.format_missing_with_indent(source!(self, item_span).lo());
         // do not take into account the lines with attributes as part of the skipped range
         let attrs_end = attrs
             .iter()
             .map(|attr| self.source_map.lookup_char_pos(attr.span().hi()).line)
             .max()
             .unwrap_or(1);
-        // Add 1 to get the line past the last attribute
-        let lo = attrs_end + 1;
+        let first_line = self.source_map.lookup_char_pos(main_span.lo()).line;
+        // Statement can start after some newlines and/or spaces
+        // or it can be on the same line as the last attribute.
+        // So here we need to take a minimum between two.
```

Comment:
```suggestion
        // So here we need to take a minimum between the two.
```

Reply:
Ty for feedback! English is not my native :)
Fixed both issues.

Reply:
nor mine ;o) anyway the comment is pretty clear so thanks for that!